dali/pipeline/operators/reader/loader/lmdb.h

```diff
@@ -24,71 +24,153 @@
 
 namespace dali {
 
-#define CHECK_LMDB(status, filename) \
+#define CHECK_LMDB(status) \
   do { \
     DALI_ENFORCE(status == MDB_SUCCESS, "LMDB Error: " + string(mdb_strerror(status)) + \
-                                        ", with file: " + filename); \
+                                        ", with file: " + db_path_); \
   } while (0)
 
-namespace lmdb {
-  inline bool SeekLMDB(MDB_cursor* cursor, MDB_cursor_op op, MDB_val* key, MDB_val *value,
-                       const string &filename) {
-    int status = mdb_cursor_get(cursor, key, value, op);
 
-    if (status == MDB_NOTFOUND) {
-      // reached the end of the db
-      return false;
+class IndexedLMDB {
+  MDB_env* mdb_env_ = nullptr;
+  MDB_cursor* mdb_cursor_ = nullptr;
+  MDB_dbi mdb_dbi_;
+  MDB_txn* mdb_transaction_ = nullptr;
+  int num_;
+  Index mdb_index_;
+  std::string db_path_;
+  Index mdb_size_;
+
+ public:
+  bool Open(const std::string& path, int num) {
+    db_path_ = path;
+    num_ = num;
+    CHECK_LMDB(mdb_env_create(&mdb_env_));
```

Comment:
I think we should check that there is no open environment at that point. Something like this:
```
DALI_ENFORCE(mdb_env_ == nullptr, 
    "Previous MDB environment was not closed")
```
Same applies to other resources that are normally released during `Close` (cursor, transaction, env)